### PR TITLE
Image Cloudinary upload

### PR DESCRIPTION
**Pull request by Dev-AkshatGupta**
Files needed to be reviewed 
Pages/SettingPage
Pages/PostPage
Component/PostInput
Pages/HomePage

Please review the UI earlier It was not reviewed .

Ignore other files Mostly CSS and controllers changes were made in them

### CODE REVIEW

File: src/Pages/PostsPage/PostPage.jsx [hunk 1]
```diff
@@ -29,29 +28,70 @@ const PostPage = () => {
...
+       data.append("cloud_name", "piston");
+       data.append("upload_preset", "fridayaaa");
+
+       fetch("https://api.cloudinary.com/v1_1/piston/image/upload" ?? "", {
+         method: "post",
+         body: data,
+       })
+         .then((res) => res.json())
+         .then((data) => {
+           console.log(data.secure_url);
```
Comment (ApurvaSawant11) [inline src/Pages/PostsPage/PostPage.jsx]: You can remove this

File: src/Pages/HomePage/HomePage.jsx [hunk 2]
```diff
@@ -13,23 +13,56 @@ function HomePage() {
...
+      fetch("https://api.cloudinary.com/v1_1/piston/image/upload" ?? "", {
+        method: "post",
+        body: data,
+      })
+        .then((res) => {
+     
+          return res.json()})
+        .then((data) => {
+           console.log(data);
+          console.log(data.secure_url);
```
Comment (ApurvaSawant11) [inline src/Pages/HomePage/HomePage.jsx]: consoles

File: src/Pages/SettingsPage/SettingsPage.jsx [hunk 3]
```diff
@@ -56,24 +84,24 @@ const SettingsPage = () => {
...
-                    type="text"
-                    name="username"
+                    type="file"
+                    accept="image/*"
+                    name="profile Photo"
                     className="w-full bg-gray-100 bg-opacity-50 rounded border border-gray-300 focus:border-indigo-500 focus:bg-white focus:ring-2 focus:ring-indigo-200 text-base outline-none text-gray-700 py-1 px-3 leading-8 transition-colors duration-200 ease-in-out"
-                    onChange={(e) =>
-                      setDetails({ ...details, username: e.target.value })
-                    }
+                    // onChange={(e) => setDetails({...details, coverPhoto:{...profilePhoto,chosen: e.target.files[0] }})}
```
Comment (ApurvaSawant11) [inline src/Pages/SettingsPage/SettingsPage.jsx]: commented code

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me! Just have a few suggestions.

1. Whenever I create a new post, the textarea does not reset. 
2. Great job with Cloudinary. But see if you can extract the imageUpload function as you are using it at 3 places. 
3. If you have time, you can also preview the image when the user uploads it. Or at least put a toast to notify the user that the image has been uploaded.
4.  Change the color of the settings dropdown. It's not visible. 
image
5. Increase the textarea height. I think textarea should have height till the `Vroom` button. Right now the empty space doesn't look good.
image

**Review by foolhardy21 — APPROVED** [review]
amazing ui. the code looks good. Just look at the suggestion given in previous review.

File: src/Components/PostInput/PostInput.jsx [hunk 4]
```diff
@@ -25,8 +26,27 @@ function PostInput({ children, userObj, setPost, post }) {
 
       <div className="postInput__bottom">
         <div className="postInput__bottom-firstBlock">
-          <GrGallery />
-          <AiOutlineFileGif />
+          <label htmlFor="post image">
```
Comment (hsnice16) [inline src/Components/PostInput/PostInput.jsx:29]: Here, you have given the `htmlFor` attribute but have not given any `id` to the `input`, so either give `id` to the `input` or remove this.

I would suggest giving `id` to the `input`. And I think, words in the `id` should be hyphenated not spaced.

Something like, `post-image`

**Review by hsnice16 — APPROVED** [review]
Great work, man. Just added a suggestion. Good Job.

File: src/Components/PostInput/PostInput.jsx (hunk 4 above)
Comment (Dev-AkshatGupta) [reply src/Components/PostInput/PostInput.jsx:29]: Thanks a lot for reviewing 
I will remove htmlFor then TBH Id is of no use for me so will remove that thanks for suggestion though

File: src/Pages/SettingsPage/SettingsPage.jsx (hunk 3 above)
Comment (Dev-AkshatGupta) [reply src/Pages/SettingsPage/SettingsPage.jsx]: Thanks for such a detailed and amazing review apurva will work on every single point you asked for thanks again

**Comment by Dev-AkshatGupta** [comment]
> amazing ui. the code looks good. Just look at the suggestion given in previous review.

Thanks for reviewing man.

**Pull request merged**